runtime/autoload/man.vim

```diff
@@ -135,7 +135,7 @@ function! s:get_page(path) abort
   let manwidth = empty($MANWIDTH) ? winwidth(0) : $MANWIDTH
   " Force MANPAGER=cat to ensure Vim is not recursively invoked (by man-db).
   " http://comments.gmane.org/gmane.editors.vim.devel/29085
-  return s:system(['env', 'MANPAGER=cat', 'MANWIDTH='.manwidth, 'man', a:path])
+  return s:system(['env', 'MANPAGER=cat', 'MANWIDTH='.manwidth, 'man', '-l', a:path])
```

Comment:
`man -l` does not work on macOS. Since both man-db and mandoc support this flag fine, we just need to add a special case to not send the flag on macOS.

Aside from that, I'm not aware of any other manpage distributions that this flag should be tested on. So we should be good.